src/pages/material.page.tsx

```diff
@@ -7,10 +9,40 @@ import '@etchteam/diamond-ui/canvas/Section/Section';
 import '@/components/composition/Wrap/Wrap';
 import '@/components/control/LocationInput/LocationInput';
 
+import LocatorApi from '@/lib/LocatorApi';
 import StartLayout from '@/pages/start.layout';
+import { MaterialWithCategory } from '@/types/locatorApi';
 
 import LocationForm from './LocationForm';
 
+async function getMaterialsOrCategoryNameById(
+  materials: string,
+  category: string,
+): Promise<string> {
+  try {
+    const materialData =
+      await LocatorApi.get<MaterialWithCategory[]>('materials');
+
+    if (category) {
+      const materialWithCategory = materialData.find(
```

Comment:
It could be, but I find it more readable this way, mainly it's weird how we're searching for a category but getting a material then finding the materials category name.

Reply:
I've updated the var names